openapi/generator_test.go

```diff
@@ -741,14 +741,20 @@ func (c customUnit) ParseExample(v string) (interface{}, error) {
 	return fmt.Sprintf("%s USD", v), nil
 }
 
-type customTime time.Time
+type customTime struct {
```

Comment:
This can stay as `type customTime time.Time`, casting it to `customTime` with `customTime{t1}` should work.

Reply:
Yeah this works. But the `String` method then needs an extra cast (as `c.Format` does not work):
```
func (c customTime) String() string {
	return time.Time(c).Format("2006-01-02T15:04:05")
}
```

Reply:
Ohhh, I see what you mean. We had a bit different ideas on how this should work. My initial ideas was that the `ParseExample` would already return the example value that can go into JSON on YAML directly (hence I also wrote some of the initial tests that way). That would be a formatted string in this case. 

The way you are thinking about this is slightly different: `ParseExample` returns a struct and then struct gets transformed into the final JSON or YAML representation through `MarshalJSON()` and `MarshalYAML()` methods, right?

Let me refactor the tests and add a note into the README.md.

Reply:
> ParseExample returns a struct 

Not specifically a struct, it returns a value which type is `customTime` (in this case), but that could be any concrete type, and yes the marshaling to JSON is implemented on that type.